src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,6 +208,29 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
+     * Returns a {@link Statement}: notify all non-overridden {@code @Ignore} methods on this class
+     * and superclasses before executing {@code next}.
+     * @param next The next statement, or {@code null}, if there is no next statement.
+     * @param notifier The runner notifier.
+     */
+    protected Statement withIgnores(final Statement next, final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                for (final T each : getFilteredChildren()) {
```

Comment:
@OpenHelios while I love Guava, we don't want to require a new third-party library for all users of JUnit.

As I said before, for backwards compatibility reasons, I think we should call the notifier via the scheduler. Unless there are a lot of ignored methods, the performance overhead of using the scheduler for notification of ignored tests shouldn't be noticeable. If you could show data that indicates otherwise, we can reconsider that.